sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs

```diff
@@ -115,6 +116,37 @@ private HttpMessage CreateRequestMessage()
             return message;
         }
 
+        private RequestUriBuilder CreateRequestUri()
+        {
+            var uri = new RequestUriBuilder();
+            uri.Reset(new Uri(_resourceEndpoint));
+
+            var (endpoint, apiVersion) = DetermineEndpointAndApiVersion();
+            uri.AppendPath(endpoint, false);
+            uri.AppendQuery("api-version", apiVersion, true);
+            return uri;
+        }
+
+        private (string Endpoint, string ApiVersion) DetermineEndpointAndApiVersion()
+        {
+            if (_scopes == null || !_scopes.Any())
+            {
+                throw new ArgumentException($"Scopes validation failed. Ensure all scopes start with either {TeamsExtentionsScopePrefix} or {EntraScopePrefix}.", nameof(_scopes));
+            }
+            else if (_scopes.All(item => item.StartsWith(TeamsExtentionsScopePrefix)))
+            {
+                return ("/access/teamsPhone/:exchangeTeamsAccessToken", "2025-03-02-preview");
```

Comment:
let's extract endpoint and version to constants, either local to this function or in the class